ui/apps/platform/cypress/integration/policies/policyWizardStep3.test.js

```diff
@@ -11,51 +10,44 @@ import {
 } from '../../helpers/policies';
 import { closeModalByButton } from '../../helpers/modal';
 import { hasFeatureFlag } from '../../helpers/features';
-import { getInputByLabel } from '../../helpers/formHelpers';
-
-const dataTransfer = new DndSimulatorDataTransfer();
-
-function dragFieldIntoSection(fieldSelector) {
-    cy.get(fieldSelector)
-        .trigger('mousedown', {
-            which: 1,
-        })
-        .trigger('dragstart', {
-            dataTransfer,
-        })
-        .trigger('drag');
-    cy.get(selectors.step3.policySection.dropTarget)
-        .trigger('dragover', {
-            dataTransfer,
-        })
-        .trigger('drop', {
-            dataTransfer,
-        })
-        .trigger('dragend', {
-            dataTransfer,
-        })
-        .trigger('mouseup', {
-            which: 1,
-        });
-}
 
-function addPolicyFieldCard(index) {
-    cy.get(selectors.step3.policyCriteria.key)
-        .eq(index)
-        .trigger('mousedown', { which: 1 })
-        .trigger('dragstart', { dataTransfer })
-        .trigger('drag');
-    cy.get(selectors.step3.policySection.dropTarget)
-        .trigger('dragover', { dataTransfer })
-        .trigger('drop', { dataTransfer })
-        .trigger('dragend', { dataTransfer })
-        .trigger('mouseup', { which: 1 });
-}
+// open Policy Fields modal, select given field, and add it to the section card
+function addPolicyField(fieldName) {
+    cy.get('.policy-section-card button:contains("Add policy field")').click();
+
+    // Note: we only use the first word of the field name to filter, because the PatternFly 4
+    // TreeView search field has a bug where it doesn't accept spaces
+    const firstWordOfFieldName = fieldName.split(' ')[0];
+    cy.log(firstWordOfFieldName);
+    cy.get('.pf-c-tree-view__search input[name="search-input"]').type(firstWordOfFieldName);
 
-function clickPolicyKeyGroup(categoryName) {
     cy.get(
-        `${selectors.step3.policyCriteria.keyGroup}:contains(${categoryName}) .pf-c-expandable-section__toggle`
+        `.pf-c-tree-view__list-item .pf-c-tree-view__list-item .pf-c-tree-view__node-title:contains(${fieldName})`
     ).click();
+
+    cy.get('.pf-c-tree-view__list-item .pf-c-tree-view__list-item .pf-c-tree-view__node').should(
+        'have.class',
+        'pf-m-current'
+    );
+
+    closeModalByButton('Add policy field');
+}
+
+// open Policy Fields modal and ensure a given field is not available
+function assertPolicyFieldNotAvailable(fieldName) {
+    cy.get('.policy-section-card button:contains("Add policy field")').click();
+
+    // Note: we only use the first word of the field name to filter, because the PatternFly 4
+    // TreeView search field has a bug where it doesn't accept spaces
+    const firstWordOfFieldName = fieldName.split(' ')[0];
+    cy.log(firstWordOfFieldName);
+    cy.get('.pf-c-tree-view__search input[name="search-input"]').type(firstWordOfFieldName);
+
+    cy.get(
```

Comment:
Some of these selectors are very verbose and might be more readable if extracted to a variable. (It would be cool to have a pf-cypress library that provided these in a semantic way, but that is a project for the future!)

Reply:
I agree that the "selector" pattern we have been using for Cypress is good for reusability, but bad for reading the code later, because you have to keep jumping between different files. For now, I will extract multiple values into local-to-this-file variables to help with maintainability.